src/server/game/Entities/Player/Player.cpp

```diff
@@ -20537,6 +20537,12 @@ void Player::ResetContestedPvP()
 
 void Player::UpdatePvPFlag(time_t currTime)
 {
+    if (!IsPvP())
+        return;
+
+    if (!pvpInfo.EndTimer || currTime < pvpInfo.EndTimer || pvpInfo.IsHostile)
+        return;
```

Comment:
these checks make most of the checks after this lines redundant. could you clean up the function a bit (while keeping the fix) ?

Reply:
This is no go, need more work.

Reply:
rechecked and commited changes 8 days ago